diff -Naur openssl-0.9.7d/crypto/rand/rand_unix.c openssl-0.9.7d.patched/crypto/rand/rand_unix.c
--- openssl-0.9.7d/crypto/rand/rand_unix.c	Sat Dec 27 08:01:52 2003
+++ openssl-0.9.7d.patched/crypto/rand/rand_unix.c	Thu Jul  4 20:55:58 2019
@@ -109,6 +109,7 @@
  *
  */
 
+#define O_NONBLOCK 00004
 #define USE_SOCKETS
 #include "e_os.h"
 #include "cryptlib.h"
diff -Naur openssl-0.9.7d/crypto/rand/randfile.c openssl-0.9.7d.patched/crypto/rand/randfile.c
--- openssl-0.9.7d/crypto/rand/randfile.c	Wed Jan 28 18:55:47 2004
+++ openssl-0.9.7d.patched/crypto/rand/randfile.c	Thu Jul  4 20:56:10 2019
@@ -166,6 +166,7 @@
 	}
 
 #if defined(O_CREAT) && !defined(OPENSSL_SYS_WIN32)
+{
 	/* For some reason Win32 can't write to files created this way */
 	
 	/* chmod(..., 0600) is too late to protect the file,
@@ -173,6 +174,7 @@
 	int fd = open(file, O_CREAT, 0600);
 	if (fd != -1)
 		out = fdopen(fd, "wb");
+}
 #endif
 	if (out == NULL)
 		out = fopen(file,"wb");
